File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        lo = 0
        hi = n-1
        while lo <= hi:
            m = (lo + hi) // 2
            if nums[m] == target:
                return m
            if nums[m] < target:
                lo = m+1
            else:
                hi= m-1
        return -1
        
            

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        int lo = 0;
        int hi = n-1;

        while (lo<=hi) {
            int m = (lo +hi)/2;
            if (nums[m] == target) {
                return m;
            }
            if (nums[m] <= target){
                lo = m+1;
            }
            else {
                hi=m-1;
            }
        }
        return -1;
        
    }
}